src/huggingface_hub/keras_mixin.py

```diff
@@ -109,10 +112,17 @@ def _create_model_card(
     if plot_model and is_graphviz_available() and is_pydot_available():
         _plot_network(model, repo_dir)
     readme_path = f"{repo_dir}/README.md"
-    model_card = "---\n"
+    metadata = {}
+    if isinstance(tags, list):
+        metadata["tags"] = tags
+    elif isinstance(tags, str):
+        metadata["tags"] = [tags]
     if task_name is not None:
-        model_card += f"tags:\n- {task_name}\n"
-    model_card += "library_name: keras\n---\n"
+        metadata["tags"].append(task_name)
```

Comment:
And I would instead do this in the upstream methods (e.g. `save_pretrained_keras`)